controllers/controllers.go

```diff
@@ -56,6 +57,39 @@ func (r *instanceReconciler) ensureInstanceIsUpToDate(instanceInfo *instance.Inf
 		return fmt.Errorf("instance cannot be nil")
 	}
 
+	nc, err := nodeconfig.NewNodeConfig(r.client, r.k8sclientset, r.clusterServiceCIDR, r.watchNamespace,
+		instanceInfo, r.signer, labelsToApply, annotationsToApply, r.platform)
+	if err != nil {
+		return fmt.Errorf("failed to create new nodeconfig: %w", err)
+	}
+
+	if instanceInfo.Node != nil {
+		// If any of the env vars changed, reboot instance so Windows services can pick up new values from OS registry
+		if _, ok := instanceInfo.Node.GetAnnotations()[metadata.InstanceRebootAnnotation]; ok {
+			// Cordon and drain the Node before we restart the instance
+			drainer := nc.NewDrainHelper()
+			if err := drain.RunCordonOrUncordon(drainer, instanceInfo.Node, true); err != nil {
+				return fmt.Errorf("unable to cordon node %s: %w", instanceInfo.Node.Name, err)
+			}
+			if err := drain.RunNodeDrain(drainer, instanceInfo.Node.Name); err != nil {
+				return fmt.Errorf("unable to drain node %s: %w", instanceInfo.Node.Name, err)
+			}
+
+			restartCmd := "Restart-Computer -Force"
+			out, err := nc.Windows.Run(restartCmd, true)
+			if err != nil {
+				return fmt.Errorf("'%s' failed with out: %s, err: %w", restartCmd, out, err)
+			}
+
+			if err := drain.RunCordonOrUncordon(drainer, instanceInfo.Node, false); err != nil {
+				return fmt.Errorf("unable to uncordon node %s: %w", instanceInfo.Node.Name, err)
+			}
+			if err := metadata.RemoveRebootAnnotation(context.TODO(), r.client, *instanceInfo.Node); err != nil {
+				return err
+			}
+		}
+	}
```

Comment:
At a minimum please make this a helper method. 

I am a bit confused about when this is being called. Has WICD finished running at this point? I also don't understand how this would work if WICD updated the environment variables outside of a configuration cycle.